drivers/serial/uart_nrfx_uarte.c

```diff
@@ -850,9 +857,14 @@ static int uarte_nrfx_rx_enable(const struct device *dev, uint8_t *buf,
 	 * way but timeouts would always occur later than expected,  most likely
 	 * after ~3 ticks.
 	 */
+#ifdef UARTE_HW_ASYNC
```

Comment:
Do you see any gain by doing that? See comment above:
```
Note that lower value would work in a similar way but timeouts would always occur later than expected,  most likely after ~3 ticks.
```
If  timeout is less than 3 RTC ticks algorithm  for setting RTC timeout (in nrf_rtc_timer.c) will fail on first try and be repeated which will lead to 3 tick timeout or more so it's better to immediately start with 3 ticks.

Reply:
With my changes I no longer receive spurious `UART_RX_RDY` events.  An explanation about why this happens is given in [ticket 62828](https://github.com/zephyrproject-rtos/zephyr/issues/62828#event-10477372536)

In summary when `rx_timeout` is set to a low value it is possible to trigger the `UART_RX_RDY` event incorrectly due to how `rx_timeout_slab` gets set.  I'm not sure if the consideration needs to be made when setting `rx_timeout_slab` when HW timers are used.

Reply:
> If timeout is less than 3 RTC ticks algorithm for setting RTC timeout (in nrf_rtc_timer.c) will fail on first try and be repeated which will lead to 3 tick timeout or more so it's better to immediately start with 3 ticks.

@nordic-krch This can happen, but not necessarily always. When kernel realizes that the tick at which the timeout is to be scheduled has already passed, it does not set up a system timer timeout for it, but it calls the handler immediately (thus, the comment you pointed out is not entirely accurate). We should let the kernel cope with the timeouts needed by the UART driver, not try to do a part of its job here. BTW, this rounding up code is anyway incorrect, as it uses `CONFIG_SYS_CLOCK_TICKS_PER_SEC` instead of `CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC`, and these two values may not be the same.